oak/server/rust/oak_runtime/src/node/grpc/client.rs

```diff
@@ -137,37 +136,79 @@ impl Node for GrpcClientNode {
 
 /// Sends gRPC requests to an external gRPC service and returns gRPC responses.
 struct GrpcRequestHandler {
+    /// The URI component of a gRPC server endpoint. Must contain the "Host" element.
+    /// https://docs.rs/tonic/0.2.1/tonic/transport/struct.Uri.html
+    uri: Uri,
+    /// Loaded PEM encoded X.509 TLS root certificate file used to authenticate an external gRPC
+    /// service.
+    root_tls_certificate: Certificate,
     /// A gRPC client dispatcher that wraps a gRPC connection and encodes/decodes messages via a
-    /// provided codec.
-    dispatcher: tonic::client::Grpc<tonic::transport::channel::Channel>,
+    /// provided codec. The value is assigned using [`GrpcRequestHandler::connect`] upon receiving
+    /// the first gRPC request.
+    dispatcher: Option<tonic::client::Grpc<tonic::transport::channel::Channel>>,
 }
 
 impl GrpcRequestHandler {
-    fn new(connection: tonic::transport::channel::Channel) -> Self {
+    fn new(uri: Uri, root_tls_certificate: Certificate) -> Self {
         Self {
-            dispatcher: tonic::client::Grpc::new(connection),
+            uri,
+            root_tls_certificate,
+            dispatcher: None,
         }
     }
 
+    /// Creates a TLS connection to an external gRPC service.
+    async fn connect(&mut self) -> Result<(), OakStatus> {
+        debug!("Connecting to {}", self.uri);
+
+        // Create a TLS configuration.
+        let tls_config = ClientTlsConfig::new().ca_certificate(self.root_tls_certificate.clone());
+
+        // Connect to a remote gRPC service.
+        let connection = Channel::builder(self.uri.clone())
+            .tls_config(tls_config)
+            .connect()
+            .await
+            .map_err(|error| {
+                error!("Couldn't connect to {}: {:?}", self.uri, error);
+                OakStatus::ErrInternal
+            })?;
+
+        self.dispatcher = Some(tonic::client::Grpc::new(connection));
+        debug!("Connected to {}", self.uri);
+        Ok(())
+    }
+
     /// Checks whether the [`GrpcRequestHandler::dispatcher`] is able to accept new requests.
     async fn ready(&mut self) -> Result<(), OakStatus> {
-        self.dispatcher.ready().await.map_err(|error| {
-            error!("Service was not ready: {}", error);
-            OakStatus::ErrInternal
-        })?;
+        self.dispatcher
+            .as_mut()
+            .expect("It not connected to service")
```

Comment:
What is "It" here?

Reply:
Changed to `Handler is not connected to a gRPC service`